### ISSUE TITLE
[Question] Common Rules -> Filter/ Decorate doesn't seem to be working. How can I debug?

### ISSUE THREAD

**Issue by alexconstantin** [querqy/querqy#386]
Hello,

I have the following rule configured:

```
glucoza =>
  FILTER: * product_ean:W13989010
```
I want to only display that product when users are searching for "glucoza", as described and exemplified here: https://opensourceconnections.com/blog/2021/06/15/pete-learns-how-to-curate-search-results-for-a-single-query/

![Screenshot 2022-11-07 at 18 54 37](https://user-images.githubusercontent.com/6921332/200368533-28c457bb-0d9a-48a8-a2c6-bd68e15bca16.png)

However, the rule doesn't get applied at all. I can successfully add Synonyms and Up/Down rules and everything works as expected, but I cannot get the Filter rule to run at all.

Also, the Decorate rule doesn't return anything as "querqy_decorations":
```
glucoza =>
  FILTER: * product_ean:W13989010
  DECORATE: REDIRECT https://www.google.com
```

I am using the latest versions of Solr, Solr4j, Querqy (querqy-solr-5.4.lucene900.1-jar-with-dependencies.jar). 

Is there something wrong with how I've declared those rules? Can someone please advice on how should I debug this issue?

P.S.: I've compared my rule files with the one generated by SMUI and the output is identical, so I am guessing it might be something related to the query. 
My solr params are these:

fl = [*]
bf = [if(termfreq(product_name_nospace,'glucoza'),100,0)]
qf = [product_name product_name_nospace product_ean]
sort = [score desc]
start = [0]
rows = [10]
defType = [querqy]
querqy.rewriters = [common_rules]
q = [glucoza]
 
Any help would be greatly appreciated. 
Thank you!

**Comment by renekrie** [comment]
Hi @alexconstantin, Just to verify: Synonyms and UP/DOWN work for exactly the same input (glucoza)?

**Comment by alexconstantin** [comment]
@renekrie yes, and also the DELETE rules (for other keywords, such as "glucoza test" -> delete "test"). It's just the FILTER/ DECORATE rules that seem to be ignored.

**Comment by alexconstantin** [comment]
The Chorus app seems to be using 2 JAR files: https://github.com/querqy/chorus/tree/main/solr/lib
Are both of them required?

**Comment by renekrie** [comment]
Re debug: If you turn on standard Solr (`debug=true`), you should see the rewritten query as part of the standard Solr debug output and also get some additional output from the CommonRules rewriter. You will get some further output if you enable "infoLogging" - https://querqy.org/docs/querqy/solr-plugin-configuration.html#info-logging

**Comment by renekrie** [comment]
Jars in Chorus: no, the querqy-regex...jar  is only an example of a custom rewriter

**Comment by renekrie** [comment]
Without the decorate issue, I would think that there might be something wrong with the analysis of the ean field. Though it would be weird to end up with a 'match all' - the filter doesn't seem to constrain the result. It might still be worth trying out the ean query as a `fq`, independent of Querqy.
On the other hand, there still is the DECORATE issue. Is there maybe a format issue in the rules definition? Line breaks?? (though it would be the first time, that a rule doesn't get applied due to formatting issue - you'd usually see a parsing error)

**Comment by alexconstantin** [comment]
If I add &debug = true, this is what I see:

```
All rules disabled (OK):

fl = [*]
qf = [product_name product_ean]
sort = [score desc]
start = [0]
rows = [10]
defType = [querqy]
querqy.rewriters = [common_rules]
q = [glucoza]
debug = [true]


"parsedquery":"DisjunctionMaxQuery((product_name:glucoza | product_ean:glucoza))",
```

```
One synonym rule (OK):

glucoza =>
  SYNONYM: synonymValue

"parsedquery":"DisjunctionMaxQuery((product_name:glucoza | product_ean:synonymvalue | product_ean:glucoza | product_name:synonymvalue))",
```

```
One Up/Down rule (OK):

glucoza =>
  UP(30): * product_ean:W41671013

"parsedquery":"FunctionScoreQuery(FunctionScoreQuery(+(product_ean:glucoza | product_name:glucoza), scored by boost(if(query(((+product_ean:w41671013 +product_ean:w +product_ean:41671013) (+product_ean:w41671013 +product_ean:w 41671013) (+product_ean:w41671013 w +product_ean:41671013) product_ean:w41671013 w 41671013),def=0.0),const(30.0),const(1.0)))))",
```

```
Filter rule (NOT OK):

glucoza =>
  FILTER: * product_ean:W41671013

"parsedquery":"DisjunctionMaxQuery((product_ean:glucoza | product_name:glucoza))",
```

The parsed query for the Filter rule acts as if the rule doesn't exist at all.

Some notes:

I am not receiving any parsing error (or any other error) while pushing the rules to Querqy. I am using the CommonRulesConfigRequestBuilder as follows:

```
 var resp = new CommonRulesConfigRequestBuilder()
        .rules(rules)
        .ignoreCase(true)
        .boostMethod(BoostInstruction.BoostMethod.MULTIPLICATIVE)
        .rhsParser(WhiteSpaceQuerqyParserFactory.class)
        .buildSaveRequest(rewriterId)
        .process(client, coreName);

      log.info("=> resp: " + resp);
```

This outputs the following:
```
=> resp: {responseHeader={status=0, QTime=1}}
```

I can confirm the rule was correctly exported by viewing the contents of this file (from the solr docker image):

/var/solr/data/configsets/testalex/conf/querqy/rewriters/common_rules

```
{"class":"querqy.solr.rewriter.commonrules.CommonRulesRewriterFactory","config":{"querqyParser":"querqy.rewrite.commonrules.WhiteSpaceQuerqyParserFactory","boostMethod":"MULTIPLICATIVE","ignoreCase":true,"rules":"glucoza =>\n  FILTER: * product_ean:W41671013"}}
```

**Comment by renekrie** [comment]
Hey @alexconstantin, Could you please verify that `solrconfig.xml` contains a line:

`<searchComponent name="query" class="querqy.solr.QuerqyQueryComponent"/>`

- This component is where filters and decorations are applied.

**Comment by alexconstantin** [comment]
@renekrie Thank you!

I did add all 3 Querqy components to SOLR (searchComponent, queryParser and requestHandler) using the SOLR Config API: https://solr.apache.org/guide/solr/latest/configuration-guide/config-api.html, not by adding them manually to the solrconfig.xml file.

```
{
  "add-searchcomponent":{
    "name": "querqy",
    "class": "querqy.solr.QuerqyQueryComponent"
  }
}
```

This created a configoverlay.json file which contained all 3 components. However it seems this doesn't work as expected with the search component.

I've moved the declaration in solrconfig.xml, as you specified, and both Filter & Decorate rules are now working as expected.

Thank you once again for your help!
